### PR TITLE
Make simapp its own gomod [Don't review]

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #3968

A few things that need to be done

- [ ] Move the simapp directory to the root. In doing this, I ran into the following issue.

```code
go: downloading github.com/cosmos/ibc-go/v7/testing/simapp v1.0.0
github.com/cosmos/ibc-go/simapp imports
        github.com/cosmos/ibc-go/modules/capability tested by
        github.com/cosmos/ibc-go/modules/capability.test imports
        github.com/cosmos/ibc-go/v7/testing/simapp: reading github.com/cosmos/ibc-go/v7/testing/simapp/go.mod at revision v7/testing/simapp/v1.0.0: unknown revision v7/testing/simapp/v1.0.0
```

I believe this is because the capability module that is being used in the simapp module is still referencing ibc-go. Once the sdk 0.50 upgrade is done, this will no longer be the case, (thanks @damiannolan !) After that, this problem should go away.

- [ ] decouple ibc-go go.mod from simapp go.mod. Currently there is a local pin to simapp to maintain existing functionality (there is not a current tag that exists to import cleanly from, I think this can be handled post sdk-50 )

- [x] Clean up Dockerfile, I didn't investigate yet, but it looks like additional files need to be added to the docker context now to build the image correctly.

- [ ] Remove dependency on simapp module from ibc-go, this largely comes in the form of `simapp.Setup(...)` in a lot of our tests.

- [ ] Usage of `suite.chainA.GetSimApp()` throughout all our tests. We have an assumption in all of our tests that we are using a `SimApp` instance directly. Can we potentially operate on an interface?

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,10 +11,11 @@ RUN test -n "${IBC_GO_VERSION}"
...
 # files are not copied before go mod download.
-ADD internal internal
-ADD testing testing
-ADD modules modules
-ADD LICENSE LICENSE
+#ADD internal internal
+#ADD testing testing
+#ADD modules modules
+#ADD LICENSE LICENSE
+ADD . .
```
[the remark this answers is not in the document]
Comment (DimitrisJim) [reply Dockerfile:18]: can look into it, probs makefile tweaks.

[the remark this answers is not in the document]
Comment (chatton) [reply Dockerfile:18]: doesn't `make build` below create the binary?

[the remark this answers is not in the document]
Comment (DimitrisJim) [reply Dockerfile:18]: this should be finding the `main.go` in `cmd`, not the one in `testing`. Building locally, this builds the binary for that main (`build_test_matrix`) byt not the one for `simapp/simd`.

**Comment by chatton** [comment]
closings this PR, it can be used as a reference for the linked issue.

**Pull request closed without merging**